go-controller/pkg/factory/handler.go

```diff
@@ -233,7 +222,16 @@ func newQueueMap(numEventQueues uint32, wg *sync.WaitGroup, stopChan chan struct
 		stopChan: stopChan,
 	}
 	for j := 0; j < int(numEventQueues); j++ {
-		qm.queues[j] = make(chan *event, 10)
+		// TODO(trozet): tweak this later
+		// we are constrained by how fast the worker threads can pull work off the queue before we can add more items
+		// into the queue. This becomes a sensitive area when we want to enqueue all initial objects. If the capacity is
+		// too low, we are worker thread bound to be able to add more objects for initial add. If the number is too high,
+		// we are consuming too much memory.
+		//
+		// If the overall workers are too slow to keep up with the rate of adds, it will still take just as long to process
+		// all the work. However, the add handler with initial objects will finish faster with a larger queue, and then
+		// other handlers can be added without waiting.
+		qm.queues[j] = make(chan *event, 10000)
```

Comment:
A clear sign that we should avoid using this framework in cluster manager. Hoping that at node we filter out uninteresting objects fast enough.